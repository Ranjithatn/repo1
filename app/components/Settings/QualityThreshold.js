import React from 'react';

const QualityThreshold = (props) => {

	const { formatMessage, nfiq } = props;

	let nfiqSlap = 0;
	let nfiqFlat = 0;
	if ( Array.isArray(nfiq) ) {
		nfiqSlap = nfiq.find( data => data.setting === "ImageQuality.SlapQualityThreshold.NFIQ" ).value;
		nfiqFlat = nfiq.find( data => data.setting === "ImageQuality.FingerQualityThreshold.NFIQ" ).value;
  }


	return (
		<div>
			<h1>{ formatMessage({ id: "NFIQQualityThreshold" }) }</h1>

			<div className="input-wrapper">
				<label>{ formatMessage({ id: "NFIQQualityThresholdSlap" }) }</label>
				<input disabled value={ nfiqSlap } className="input" />
			</div>

			<div className="input-wrapper">
				<label>{ formatMessage({ id: "NFIQQualityThresholdFlat" }) }</label>
				<input disabled value={ nfiqFlat } className="input" />
			</div>

		</div>
	)

}


export default QualityThreshold;
