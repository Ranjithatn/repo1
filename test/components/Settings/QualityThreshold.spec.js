import React from "react";
import toJson from "enzyme-to-json";
import QualityThreshold from "../../../app/components/Settings/QualityThreshold";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow, mount } from "enzyme";


configure({ adapter: new Adapter() });

const props = {
	formatMessage: data => data.id
  }

describe('<QualityThreshold />', () => {

	it("should render without crashing", () => {
		const component = mount(
			<QualityThreshold {...props}/>
		);
		const tree = toJson(component);
		expect(tree).toMatchSnapshot();
	});

});
