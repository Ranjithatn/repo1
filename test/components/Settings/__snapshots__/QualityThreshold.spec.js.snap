// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QualityThreshold /> should render without crashing 1`] = `
<QualityThreshold
  formatMessage={[Function]}
>
  <div>
    <h1>
      NFIQQualityThreshold
    </h1>
    <div
      className="input-wrapper"
    >
      <label>
        NFIQQualityThresholdSlap
      </label>
      <input
        className="input"
        disabled={true}
        value={0}
      />
    </div>
    <div
      className="input-wrapper"
    >
      <label>
        NFIQQualityThresholdFlat
      </label>
      <input
        className="input"
        disabled={true}
        value={0}
      />
    </div>
  </div>
</QualityThreshold>
`;
